### PR TITLE
test: add packaged CLI release process to CI workflow

### PR DESCRIPTION
**Pull request by edwardfoyle**
After this change, the CLI release pipeline will look like: https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/2698/workflows/4db5a237-da62-46d7-a84d-2c362dead9d3
(currently this PR just enables this workflow on the backendManager branch but it can be switched to the release branch later)

The new steps / scripts will build the packaged CLI for all platforms, generate a GitHub pre-release and populate the description with a unified changelog of all changes going out, and upload the binaries to the release.
Then the workflow will validate that the installation script can install the packaged CLI and then promote the pre-release to a full release.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by yuth — APPROVED** [review]
Can we also generate unified changelog in the repository itself so customers can see what is changed.

File: .circleci/config.base.yml [hunk 1]
```diff
@@ -503,7 +516,52 @@ jobs:
...
+            version=$(cat .amplify-pkg-version)
+            yarn ts-node scripts/github-prerelease.ts $version
 
+  github_prerelease_install_sanity_check:
+    <<: *defaults
+    steps:
+      - restore_cache:
+          key: amplfiy-pkg-tag-{{ .Branch }}-{{ .Revision }}
+      - run:
+          name: Install packaged Amplify CLI
```
[the remark this answers is not in the document]
Comment (edwardfoyle) [reply .circleci/config.base.yml:542]: assuming your talking about the line below, yes good catch, I need to replace that

**Comment by edwardfoyle** [comment]
@yuth you think the unified changelog should be committed to the repository? I can do that, but customers can click on the release on the repo page to see the changelog

**Comment by yuth** [comment]
> @yuth you think the unified changelog should be committed to the repository? I can do that, but customers can click on the release on the repo page to see the changelog

I would prefer to have changelog at the root of the project as well which is easier to find

**Review by kaustavghosh06 — APPROVED** [review]

**Comment by edwardfoyle** [comment]
@yuth My hesitation with doing that is
1. The unified changelog that this PR is generating for the release does not have changes from previous versions so it's more like "release notes" rather than a changelog in that regard
2. All of the information in the unified changelog is already committed in the individual package's changelogs so putting it all at the top level would be duplicating data that already there. I realize it's easier to grok when it's all together, but that's what the release notes are for

I think for this PR I'm going to leave it as-is (mainly because it takes a long time to test out new scripts on CircleCI) but I'm open to iterating on the unified changelog approach.

**Pull request merged**